Dockerfile

```diff
@@ -1,10 +1,11 @@
 # setup docker file for the project
-FROM  node:16.9.0
+FROM  node:16-alpine3.12
 
 WORKDIR /app
 COPY package*.json ./
-RUN npm ci --only=production
-
+RUN npm install
 COPY . .
+RUN npm run build
+RUN npm ci --only=production
```

Comment:
You may not need to do `npm install` when you have `npm ci`. Checkout this article: https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable

Reply:
The article is resourceful I have taken action. I wanted to ensure that all dependencies including the dev-dependencies (`Useful in building the application`) but then after they can be removed in such case. 
In case we prune the `dev-dependencies` the build will fail

Reply:
Well, that would mean a bigger problem if removing them cause a failing build because as the name suggests, they are only supposed to be used in development; if we move production, their absence shouldn't cause any issue.

Reply:
You  are right, but since the build process is considered a development stage. That is where the need arises. 
You are right, removing them does not cause the issue, but that would mean ensuring that the user run a build and then after use docker which means that we would add the `dist` folder on tracking since it would not be available otherwise on the host when using version control.

These dependencies including Typescript will only be used in `build` and can be removed afterwards